adapters/sovrnXsp/sovrnXsp.go

```diff
@@ -0,0 +1,167 @@
+			appCopy.Publisher = &publisherCopy
+		}
+		request.App = &appCopy
+	} else {
+		return nil, []error{&errortypes.BadInput{
+			Message: "non-app request",
+		}}
+	}
+
+	var errors []error
+	var imps []openrtb2.Imp
+
+	for idx, imp := range request.Imp {
+		if imp.Banner == nil && imp.Video == nil && imp.Native == nil {
+			continue
+		}
+
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: ext.bidder not provided", idx),
+			}
+			errors = append(errors, err)
+			continue
+		}
+
+		var xspExt openrtb_ext.ExtImpSovrnXsp
+		if err := json.Unmarshal(bidderExt.Bidder, &xspExt); err != nil {
+			err = &errortypes.BadInput{
+				Message: fmt.Sprintf("imp #%d: %s", idx, err.Error()),
+			}
+			errors = append(errors, err)
+			continue
+		}
+
+		if xspExt.MedID != "" {
+			request.App.ID = xspExt.MedID
+		}
+		if xspExt.PubID != "" {
+			request.App.Publisher.ID = xspExt.PubID
+		}
+		if xspExt.ZoneID != "" {
+			imp.TagID = xspExt.ZoneID
+		}
+		imps = append(imps, imp)
+	}
+
+	if len(imps) == 0 {
+		return nil, errors
+	}
+
+	request.Imp = imps
+	requestJson, err := json.Marshal(request)
+	if err != nil {
+		return nil, append(errors, err)
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     a.Endpoint,
+		Body:    requestJson,
+		Headers: headers,
+	}}, errors
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+	if responseData.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d", responseData.StatusCode),
+		}}
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	var errors []error
+	result := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+
+	for _, seatBid := range response.SeatBid {
+		for _, bid := range seatBid.Bid {
+			bid := bid
+			var ext bidExt
+			if err := json.Unmarshal(bid.Ext, &ext); err != nil {
+				errors = append(errors, err)
+				continue
+			}
+
+			var bidType openrtb_ext.BidType
+			switch ext.CreativeType {
+			case creativeTypeBanner:
+				bidType = openrtb_ext.BidTypeBanner
+			case creativeTypeVideo:
+				bidType = openrtb_ext.BidTypeVideo
+			case creativeTypeNative:
+				bidType = openrtb_ext.BidTypeNative
+			default:
+				errors = append(errors, &errortypes.BadServerResponse{
+					Message: fmt.Sprintf("Unsupported creative type: %d", ext.CreativeType),
+				})
+			}
+
+			bid.Ext = nil
```

Comment:
Any reason to set `bid.Ext` to nil

Reply:
Since `bid.Ext` is unlikely to contain anything relevant to the upstream auction flow. Removed the `nil` assignment in [1].

[1] https://github.com/prebid/prebid-server/pull/3312/commits/ae2e1090252bee341795f2ccbf3dcb4cb3641422